lib/internal/process/parse_args/utils.js

```diff
@@ -0,0 +1,160 @@
+ * @example
+ * isOptionValue('V']) // returns true
+ * isOptionValue('-v') // returns false
+ * isOptionValue('--foo') // returns false
+ * isOptionValue(undefined) // returns false
+ */
+function isOptionValue(value) {
+  if (value == null) return false;
+  if (value === '-') return true; // e.g. representing stdin/stdout for file
+
+  // Open Group Utility Conventions are that an option-argument
+  // is the argument after the option, and may start with a dash.
+  // However, we are currently rejecting these and prioritising the
+  // option-like appearance of the argument. Rejection allows more error
+  // detection for strict:true, but comes at the cost of rejecting intended
+  // values starting with a dash, especially negative numbers.
+  return !StringPrototypeStartsWith(value, '-');
+}
+
+/**
+ * Determines if `arg` is a just a short option.
+ * @example '-f'
+ */
+function isLoneShortOption(arg) {
+  return arg.length === 2 &&
+    StringPrototypeCharAt(arg, 0) === '-' &&
+    StringPrototypeCharAt(arg, 1) !== '-';
+}
+
+/**
+ * Determines if `arg` is a lone long option.
+ * @example
+ * isLoneLongOption('a') // returns false
+ * isLoneLongOption('-a') // returns false
+ * isLoneLongOption('--foo) // returns true
+ * isLoneLongOption('--foo=bar) // returns false
+ */
+function isLoneLongOption(arg) {
+  return arg.length > 2 &&
+    StringPrototypeStartsWith(arg, '--') &&
+    !StringPrototypeIncludes(StringPrototypeSlice(arg, 3), '=');
+}
+
+/**
+ * Determines if `arg` is a long option and value in same argument.
+ * @example
+ * isLongOptionAndValue('--foo) // returns false
+ * isLongOptionAndValue('--foo=bar) // returns true
+ */
+function isLongOptionAndValue(arg) {
+  return arg.length > 2 &&
+    StringPrototypeStartsWith(arg, '--') &&
+    StringPrototypeIncludes(StringPrototypeSlice(arg, 3), '=');
+}
+
+/**
+ * Determines if `arg` is a short option group.
+ *
+ * See Guideline 5 of the [Open Group Utility Conventions](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html).
+ *   One or more options without option-arguments, followed by at most one
+ *   option that takes an option-argument, should be accepted when grouped
+ *   behind one '-' delimiter.
+ * @example
+ * isShortOptionGroup('-a', {}) // returns false
+ * isShortOptionGroup('-ab', {}) // returns true
+ * // -fb is an option and a value, not a short option group
+ * isShortOptionGroup('-fb', {
+ *   options: { f: { type: 'string' }}
+ * }) // returns false
+ * isShortOptionGroup('-bf', {
+ *   options: { f: { type: 'string' }}
+ * }) // returns true
+ * // -bfb is an edge case, return true and caller sorts it out
+ * isShortOptionGroup('-bfb', {
+ *   options: { f: { type: 'string' }}
+ * }) // returns true
+ */
+function isShortOptionGroup(arg, options) {
+  if (arg.length <= 2) return false;
+  if (StringPrototypeCharAt(arg, 0) !== '-') return false;
+  if (StringPrototypeCharAt(arg, 1) === '-') return false;
```

Comment:
```suggestion
  if (StringPrototypeCharAt(arg, 0) !== '-' ||
      StringPrototypeCharAt(arg, 1) === '-') return false;
```

Reply:
I deliberately did a special case per line for the early exits. Reason about one at a time, then get onto the actual "logic" in the routine. Just cosmetic though, no objection to change if people tripping over the pattern.

Reply:
will leave as is.